src/bootstrap/setup.rs

```diff
@@ -219,6 +229,38 @@ fn try_link_toolchain(stage_path: &str) -> bool {
         .map_or(false, |output| output.status.success())
 }
 
+fn create_stage1_toolchain_placeholder_if_it_does_not_already_exist(stage_path: &str) -> bool {
+    let mut pathbuf = PathBuf::from(stage_path);
+
+    pathbuf.push("lib");
+    if fs::create_dir_all(&pathbuf).is_err() {
+        return false;
+    };
+    pathbuf.pop();
+
+    pathbuf.push("bin");
+    if fs::create_dir_all(&pathbuf).is_err() {
+        return false;
+    };
+
+    pathbuf.push(format!("rustc{}", EXE_SUFFIX));
+
+    let file_already_exists =
+        fs::metadata(&pathbuf).ok().as_ref().map(fs::Metadata::is_file) == Some(true);
```

Comment:
I doubt we need to worry about it not being a file? path.exists() should be sufficient, I think.

Reply:
I just checked that `path.exists()` returns true if the path is actually a directory and not a file. So I don't actually see harm in checking if it's a file.

Reply:
This is just more complex and not really all that more helpful -- I'd prefer to avoid the extra complexity here.

Reply:
Updated